### DISCUSSION TITLE
Remove extension default snippets

### DISCUSSION THREAD

**Discussion by hyangah** [golang/vscode-go#1142] [Discussions]
Gopls is enabled by default, and it can provide context-aware, better suggestions than snippets.
Snippets are often too opinionated, and interfere with the suggestions from the language server.
VS Code supports user snippets so users can organize the snippets to their liking.

I heard the news that the recently released python extension finally removed the default snippets.
https://github.com/microsoft/vscode-python/blob/release-2021.01/CHANGELOG.md#enhancements

Related: https://github.com/golang/vscode-go/issues/160

What do you think?

**Comment by marwan-at-work** [comment]
There are two snippets I use all the time: `pkgm` and `iferr` 

I know gopls has a really nice `if err != nil` snippet completion, but that one is only triggered on specific conditions, so as a fallback I tend to use the default `iferr` snippet completion frequently. 

I also use `tys` and `tyi` as well. 

Of course, if vscode-go removed those snippets I can just add them myself as "Custom User Snippets", so I'm happy either way. 

Maybe it's worth reducing the snippets or moving them to gopls?
